### DISCUSSION TITLE
Varying gradient by group variable

### DISCUSSION THREAD

**Discussion by mattmoo** [lindeloev/mcp#121] [General]
I know that it is possible in mcp to calculate varying changepoints by a grouping variable, but is it possible to do the same with gradients?

**Comment by lindeloev** [comment]
If you mean continuous variables, then yes, it will be in the future (and this would be unique to `mcp` AFAIK). They still have to be observed within a group, but the syntax would be something like:

```r
model = list(
    y ~ x,
    1 + (0 + age | participant) ~ 0 + x
)
```

So each `participant` has one `age` and you would get an estimate on how the change point location is affected by age. If that's what you had in mind? The grouping (`participant`) of the continuous predictor (`age`) is necessary for model identifiability, I think.

**Comment by mattmoo** [comment]
I think so, the specific situation that I have right now is to compare trajectories in patient outcomes between social groups. I guess that it would need different intercepts per group also (and zero changepoints!). It is easy enough to work around for now by fitting separate models to each group.

A future situation that I have in mind is a [stepped-wedge trial](https://www.bmj.com/content/350/bmj.h391), wherein the implementation is applied to each site at a different timepoint (already supported), but sites may also differ in intervention-agnostic factors, so some further clustering in the model would be necessary.
